pkg/providers/aws/provider_postgres.go

```diff
@@ -189,11 +190,19 @@ func (p *PostgresProvider) createRDSInstance(ctx context.Context, cr *v1alpha1.P
 	}
 
 	// create rds instance if it doesn't exist
-	if foundInstance == nil {
+	if foundInstance == nil && !annotations.Has(cr, resourceIdentifierAnnotation) {
 		logrus.Info("creating rds instance")
-		if _, err = rdsSvc.CreateDBInstance(rdsCfg); err != nil {
+		out, err := rdsSvc.CreateDBInstance(rdsCfg)
+		if err != nil {
 			return nil, croType.StatusMessage(fmt.Sprintf("error creating rds instance %s", err)), err
 		}
+
+		if out != nil && out.DBInstance != nil && out.DBInstance.DBInstanceIdentifier != nil {
```

Comment:
It might be better to error in the cases where any of these are nil? If so, I'd need to figure out how to make the unit tests work in that case, since they'd be nil by default.

Maybe I can get them from the `rdsCfg` that's passed as input instead...I'll try that :thinking: